app/src/main/java/com/example/poste/activities/EditPostActivity.java

```diff
@@ -59,84 +39,73 @@ public class EditPostActivity extends AppCompatActivity {
+
+        String postId = getIntent().getStringExtra("postID");
+        String folderId = getIntent().getStringExtra("folderID");
 
-        // Prep vars
-        String currentFolderTitle;
         Post currentPost;
-        Folder currentFolder;
-        HashMap<Folder, FolderAccess> userFolders;
-        HashMap<String, Integer> folderIdNameMap = new HashMap<>();
-        EditText postNameField = findViewById(R.id.editTextNewName);
-        EditText postLinkField = findViewById(R.id.editTextNewLink);
-        Spinner spinner = findViewById(R.id.edit_post_spinner);
-        Button cancelBtn = findViewById(R.id.edit_post_cancel_btn);
-        Button saveBtn = findViewById(R.id.edit_post_save_button);
-        try {
-            currentPost = API.getPostById(getIntent().getIntExtra("postId", 0));
-            currentFolder = API.getFolderById(getIntent().getIntExtra("folderId", 0));
-            currentFolderTitle = String.format("(%d) %s", currentFolder.getId(), currentFolder.getName());
-            userFolders = API.getFoldersForUserId(PosteApplication.getCurrentUser().getId());
-            for (Folder folder: userFolders.keySet()) {
-                folderIdNameMap.put(String.format("(%d) %s", folder.getId(), folder.getName()), folder.getId());
+        Folder currentFolder = User.getUser().getFolder(folderId);
+        if (currentFolder == null){
+            currentPost = null;
+            Log.e("Error", "onCreate: Folder not found");
+        } else {
+            currentPost = User.getUser().getFolder(folderId).getPostFromFolder(postId);
+            if (currentPost == null){
+                Log.e("Error", "onCreate: Post not found");
+            } else {
+                postTitle.setText(currentPost.getTitle());
+                postDescription.setText(currentPost.getDescription());
+                linkToPost.setText(currentPost.getUrl());
             }
-        } catch (APIException e) {
-            throw new RuntimeException(e);
         }
 
-        // Setup for the dropdown menu
-        ArrayAdapter<String> adapter = new ArrayAdapter<String>(this, android.R.layout.simple_spinner_item, new ArrayList<>(folderIdNameMap.keySet()));
-        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
-        spinner.setAdapter(adapter);
-
-        // Set the default values
-        postNameField.setText(currentPost.getName());
-        postLinkField.setText(currentPost.getLink());
-        spinner.setSelection(adapter.getPosition(currentFolderTitle));
-
-        // Cancel button click handler
-        cancelBtn.setOnClickListener(view -> {
-            // Send back to folder view
-            sendToDashboard();
+        buttonCancelChanges.setOnClickListener(view -> {
+            Intent returnToFolderIntent = new Intent(EditPostActivity.this, EditPostActivity.class);
+            finish();
+            returnToFolderIntent.putExtra("postID",postId);
```

Comment:
Ah, I was hoping for an example, thank you. This might be beneficial in the docstring too

Reply:
Has since been added, thanks to your review